easy-ui-react/src/Checkbox/Checkbox.stories.tsx

```diff
@@ -0,0 +1,136 @@
+import { Meta, StoryObj } from "@storybook/react";
+import React, { useState } from "react";
+import { VerticalStack } from "../VerticalStack";
+import { Checkbox, CheckboxProps } from "./Checkbox";
+
+type Story = StoryObj<typeof Checkbox>;
+
+const Template = (args: CheckboxProps) => <Checkbox {...args} />;
+
+const meta: Meta<typeof Checkbox> = {
+  title: "Components/Checkbox",
+  component: Checkbox,
+  args: {
+    children: "Checkbox item",
+    isDisabled: false,
+    isReadOnly: false,
+    isIndeterminate: false,
+    isNested: false,
+    size: "md",
+    validationState: "valid",
+  },
+  argTypes: {
+    children: {
+      control: "text",
+    },
+    errorText: {
+      control: "text",
+    },
+  },
+  parameters: {
+    controls: {
+      exclude: ["onChange"],
+    },
+  },
+};
+
+export default meta;
+
+export const Default: Story = {
+  render: Template.bind({}),
+};
+
+export const Selected: Story = {
+  render: Template.bind({}),
+  args: {
+    isSelected: true,
+  },
+};
+
+export const Disabled: Story = {
+  render: Template.bind({}),
+  args: {
+    isDisabled: true,
+  },
+};
+
+export const Indeterminate: Story = {
+  render: Template.bind({}),
+  args: {
+    isIndeterminate: true,
+  },
+};
+
+export const Error: Story = {
+  render: Template.bind({}),
+  args: {
+    children: "Error item",
+    validationState: "invalid",
+    errorText: "This field is required",
+  },
+};
+
+export const Nested: Story = {
+  render: () => {
+    // This is a naive implementation to showcase nesting. Use caution copying
```

Comment:
**question**: do you think its worth pursing a `CheckboxGroup` component, or some kind of component that manages nested/grouped checkboxes at some point for easy ui?